### ISSUE TITLE
Backups on external drive do not work

### ISSUE THREAD

**Issue by AnEmortalKid** [AnEmortalKid/starcitizen-byecache#9]
> Handy tool, only, I've installed it and pointed the backup folder to an external drive's dir.. but it doesn't put the backup there. Maybe it has to do with permission issues ? it should tell when it can't write a backup for some reason.

**Comment by RealCosmicD** [comment]
[main.log](https://github.com/AnEmortalKid/starcitizen-byecache/files/13029702/main.log)

I've tested this, and the only thing I see change when I click "purge user data", is that it adds a line that it deleted the user/client folder, but I don't see it adding a backup. (or attempting something in the log)

**Comment by RealCosmicD** [comment]
![image](https://github.com/AnEmortalKid/starcitizen-byecache/assets/84868665/7201939e-634d-4317-867a-d4a509ed9f82)
These are my settings

**Comment by AnEmortalKid** [comment]
Those folders already exist @RealCosmicD ? 

Also thank you for the details. I definitely need logs around all of the operations.

**Comment by RealCosmicD** [comment]
![image](https://github.com/AnEmortalKid/starcitizen-byecache/assets/84868665/da4b684a-80fa-4e76-860d-cf60aace3c19)
Yep, they do. (and dragging a file from the downloads folder to it doesn't give me an entitlement window or something...

**Comment by AnEmortalKid** [comment]
Oh I think I see the bug... I incorrectly assumed that everyone would have a `USER/Client/0/Controls/Mappings` in their backup directories (because I had one and ran into a conflict), so it skips if you don't. 

I'll fix that and test it out.

**Comment by RealCosmicD** [comment]
Ah, I guess that is created when you save mappings right? I never do that in this stage anyway, because so much of the game changes, importing mappings might cause issues in a next major patch.

**Comment by AnEmortalKid** [comment]
Yeah I wanted to save the whole folder and figured the easiest was to just copy the whole structure. Bad move we only care about the mappings files !

**Comment by AnEmortalKid** [comment]
Hey @RealCosmicD , sorry its been taking me a minute but I've been sick and thinking isn't the easiest.

I've published an updated version to this release https://github.com/AnEmortalKid/starcitizen-byecache/releases/tag/v0.1.1

Notable changes:

- The logs are more verbose, we'll find out what its trying to backup (how many files and which) and when it skips for some reason.
- The whole directory structure is no longer copied, the only thing that will end up in your backup folder will be the `xml` files, like so:

![image](https://github.com/AnEmortalKid/starcitizen-byecache/assets/3474369/35879927-a02f-4948-8107-0b0a06ed2873)


I still have to test with an external harddrive, but I'm hoping with more logging we can determine if its even trying to write or some broken logic I have doesn't even write.

For reference, here's what one of my paths points to:

![image](https://github.com/AnEmortalKid/starcitizen-byecache/assets/3474369/bd34209d-70f0-466e-a495-6fa46d2f5878)

And if that had a USER/Client/0/Controls/Mappings directory with XMLS, it would copy all those to the backup.
